azkaban-common/src/main/java/azkaban/imagemgmt/daos/RampRuleDaoImpl.java

```diff
@@ -245,6 +268,38 @@ public int addFlowDenyInfo(final List<ProjectFlow> flowIds, final String ruleNam
     }
   }
 
+  /**
+   * Update version on the rule, both table flow_deny_lists and ramp_rules got refreshed data.
+   * It will fetch ramp_rules to get metadata and operate UPDATE on both in a single transaction.
+   *
+   * @param newVersion
+   * @param ruleName - ruleName in {@see ImageRampRule}
+   * @param user
+   * @throws azkaban.imagemgmt.exception.ImageMgmtDaoException
+   */
+  @Override
+  public void updateVersionOnRule(final String newVersion, final String ruleName, final String user) {
+    final SQLTransaction<Long> fetchRampRuleAndUpdateDenyList = transOperator -> {
+      final ImageRampRule imageRampRule =
+          transOperator.query(FetchRampRuleHandler.FETCH_RAMP_RULE_BY_ID, new FetchRampRuleHandler(), ruleName);
+      final String newDenyVersion = String.join(":", imageRampRule.getImageName(), newVersion);
+      if (!newVersion.equals(imageRampRule.getImageVersion())) {
```

Comment:
In which case the newVersion will be same as existing version?

Reply:
It might be, if someone makes a mistake to put old version. Or others in the team don't know someone had already update the version to the target one, they might be update the same version.